tests/org.jboss.reddeer.eclipse.test/src/org/jboss/reddeer/eclipse/test/m2e/core/ui/preferences/MavenErrorPreferencePageTest.java

```diff
@@ -0,0 +1,82 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.reddeer.eclipse.test.m2e.core.ui.preferences;
+
+import static org.junit.Assert.assertTrue;
+
+import org.jboss.reddeer.eclipse.m2e.core.ui.preferences.MavenErrorPreferencePage;
+import org.jboss.reddeer.junit.runner.RedDeerSuite;
+import org.jboss.reddeer.swt.exception.SWTLayerException;
+import org.jboss.reddeer.workbench.ui.dialogs.WorkbenchPreferenceDialog;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test for MavenErrorPreferencePage
+ * 
+ * @author ldimaggi
+ *
+ */
+
+@RunWith(RedDeerSuite.class)
+public class MavenErrorPreferencePageTest {
+
+	private WorkbenchPreferenceDialog preferencesDialog = new WorkbenchPreferenceDialog();
+	
+	private MavenErrorPreferencePage mavenPreferencePage = new MavenErrorPreferencePage();
+
+	@Test
+	public void checkAllPreferences() {
+		preferencesDialog.open();
+		preferencesDialog.select(mavenPreferencePage);
+		
+		/* Verify default values */		
+		assertTrue(mavenPreferencePage.getGroupId().equals("Warning"));
+		assertTrue(mavenPreferencePage.getVersion().equals("Warning"));
+		assertTrue(mavenPreferencePage.getProjectConfig().equals("Error"));
+		assertTrue(mavenPreferencePage.getPluginExecution().equals("Error"));
+		assertTrue(mavenPreferencePage.getManagedVersion().equals("Warning"));
+		
+		/* Change values - verify changes are made */
+		mavenPreferencePage.setGroupId("Ignore");
+		mavenPreferencePage.setVersion("Ignore");
+		mavenPreferencePage.setProjectConfig("Ignore");
+		mavenPreferencePage.setPluginExecution("Ignore");
+		mavenPreferencePage.setManagedVersion("Ignore");
+		
+		assertTrue(mavenPreferencePage.getGroupId().equals("Ignore"));
+		assertTrue(mavenPreferencePage.getVersion().equals("Ignore"));
+		assertTrue(mavenPreferencePage.getProjectConfig().equals("Ignore"));
+		assertTrue(mavenPreferencePage.getPluginExecution().equals("Ignore"));
+		assertTrue(mavenPreferencePage.getManagedVersion().equals("Ignore"));
+	
+		/* Reset to defaults - verify values */
+		mavenPreferencePage.restoreDefaults();
+		assertTrue(mavenPreferencePage.getGroupId().equals("Warning"));
```

Comment:
asserting restored default values is not required for our tests. If it fails, it is Eclipse issue so we would be testing Eclipse there instead of RedDeer. Also there is no need to restore default if Apply button has not been clicked (and will not be once changes in code will take effect).